sx/pisa3/pisa_util.py

```diff
@@ -470,7 +468,7 @@ def __getattr__(self, name):
                      % (self.__class__.__name__, name)
             raise AttributeError(e)
 
-_rx_datauri = re.compile("^data:(?P<mime>[a-z]+/[a-z]+);base64,(?P<data>.*)$", re.M | re.DOTALL)
```

Comment:
`re.MULTILINE` makes `$` match on newlines.  Precisely because URIs may contain whitespace, we want `$` to match only at the end of the string.  There is a greedy `.*` before it, so `$` won't actually match a newline in the middle of the string.

Similarly for `^`, with `re.MULTILINE` this might actually match starting from a newline in the middle of the string, however we use `re.match()` on this pattern, forcing the `^` to only match the start of the string.

Leaving the `re.MULTILINE` there made me uneasy, because the only reason that `re.MULTILINE` *isn't* causing bugs are these other conditions that mask the `re.MULTILINE` behavior.